### ISSUE TITLE
Bug: Jacobian of Jacobian in pytorch produces null matrices

### ISSUE THREAD

**Issue by tramy1258** [geomstats/geomstats#1801]
### Describe the bug

Computing `inner_product_derivative_matrix` in `FisherRaoMetric` requires computing Jacobian and Hessian of `pdf` at given base points. We would like to use Jacobian of Jacobian to compute Hessian matrix to be more time efficient, instead of computing Hessian and Jacobian separately. This method works in `autograd` but not in `pytorch` where outputs are just null matrices. The error can be reproduced with the code below. 

Another point that might be worth looking into is that using nested Jacobian produces matrix of shape `(...,1,dim,dim)` whereas using Hessian produces matrix of shape `(...,dim,dim)`.

### Steps/Code to Reproduce

```
import geomstats as gs

def test(x):
    return gs.array([gs.sum(x**4)])

# Basic case
point = gs.array([0.1,0.5])
jj = gs.autodiff.jacobian(gs.autodiff.jacobian(test))(point)
print(jj,jj.shape)
print('-'*30)

# Vectorized case
point = gs.array([[0.1,0.5],[0.2,0.7]])
jj = gs.autodiff.jacobian_vec(gs.autodiff.jacobian_vec(test))(point)
print(jj,jj.shape)
```

### Expected Behaviour

Similar behaviour to `autograd`.

### Actual Behaviour

Outputs null matrices.

### Your environment

```shell
* pytorch, autograd
* Windows 10
```

**Comment by yangshengaa** [comment]
From my understanding, currently we are not able to differentiate on top of `gs.autodiff.jacobian_vec` because by default pytorch `TORCH.AUTOGRAD.FUNCTIONAL.JACOBIAN(...)` puts `create_graph=False`, which means the gradient infomation is not tracked when computing the first order, and thus cannot generate second order when calling it again. 

Below is the source code from [geomstats._backend.pytorch.autodiff](https://github.com/geomstats/geomstats/blob/main/geomstats/_backend/pytorch/autodiff.py):

```
def jacobian_vec(func, point_ndim=1):
    """<some docstrings>
    """

    def _jacobian(point):
        if point.ndim == point_ndim:
            return _torch_jacobian(func=lambda x: func(x), inputs=point)
        return _torch.stack(
            [
                _torch_jacobian(func=lambda x: func(x), inputs=one_point)
                for one_point in point
            ],
            axis=0,
        )

    return _jacobian
```

you see that `create_graph=True` is not there.

For now you may just copy this chunk of code into your code base and add the `create_graph=True` argument.
